app/src/main/java/com/example/films/MainActivity.kt

```diff
@@ -16,15 +16,19 @@ class MainActivity : AppCompatActivity(),
         this.supportActionBar?.hide()
     }
 
-    override fun filmOnClick() {
+
+    override fun backOnClick() {
+        supportFragmentManager.popBackStack()
+    }
+
+    override fun movieOnClick(movie: Movie) {
+
+        val fragment = FragmentMoviesDetails()
+        fragment.movie = movie
```

Comment:
if you set movie with the method
when configuration change it's can't save the movie 
due that you need to send a movie to FragmentMoviesDetails as a fragment argument

Reply:
I solve this problem by adding movie in Bundle (but I had to implement interface Serializable in class Movie, perhaps it's not very good). In video I change configuration and all is good

I made:
class FragmentMoviesDetails(val movie: Movie)
and in movieOnClick realization (in activity): 
val fragment = FragmentMoviesDetails(movie)
supportFragmentManager.beginTransaction()
            .replace(R.id.fragment_container, fragment)
            .addToBackStack(null)
            .commit()

and when configuration change on FragmentMoviesDetails, I get this exception:
E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.example.films, PID: 31985
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.example.films/com.example.films.MainActivity}: androidx.fragment.app.Fragment$InstantiationException: Unable to instantiate fragment com.example.films.FragmentMoviesDetails: could not find Fragment constructor

(perhaps I need to send a movie to FragmentMoviesDetails as a fragment argument somehow differently, I don't understand, with 2 constructors...)

Reply:
you need set argument like so
val fragment = FragmentMoviesDetails()
val bundle = bundleOf("movie" to movie)
fragment.arguments = bundle